src/producer/vector_producer.rs

```diff
@@ -0,0 +1,396 @@
+use crate::producer::{ExtractCodeError, ResultsProducer};
+use crate::types::result::CheckResult;
+use reqwest::Client;
+use sentry_kafka_schemas::Schema;
+use tokio::sync::mpsc::{self, UnboundedReceiver, UnboundedSender};
+use tokio::task::JoinHandle;
+
+const VECTOR_BATCH_SIZE: usize = 2;
+
+pub struct VectorResultsProducer {
+    schema: Schema,
+    sender: UnboundedSender<Vec<u8>>,
+}
+
+impl VectorResultsProducer {
+    pub fn new(topic_name: &str) -> (Self, JoinHandle<()>) {
+        Self::new_with_endpoint(topic_name, "http://localhost:8020".to_string())
+    }
```

Comment:
IMO this should not be part of the public interface since it's just for testing. Maybe rename `new_with_endpoint` to just `new` and have it always take the vector endpoint.

Alternatively you can setup a public `VectorResultsProducerConfig` struct hat has these two things in it